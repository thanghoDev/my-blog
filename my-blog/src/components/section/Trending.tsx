import React from 'react';
import Link from 'next/link';
import { Col } from 'react-bootstrap';

import Styles from '@/styles/Trending.module.css';
import { Blog } from '@/types/blog';

type TrendingProps = {
  trending: Blog[];
};
function Trending({ trending }: TrendingProps) {
  return (
    <Col md={4} className='mt-5 ps-4'>
      <h2 className='mb-5 text-decoration-underline fs-5'>Trending</h2>
      {trending.map((item, index) => (
        <div key={item.id} className='mt-3 d-flex'>
          <h3 className={Styles.number}>0{index + 1}</h3>
          <div className='ms-4'>
            <h3 className={`text-black text-capitalize ${Styles.title}`}>
              {item.title}
            </h3>
            <p>
              Dave Rogers <code className='text-secondary'>in</code> Trending
            </p>
          </div>
        </div>
      ))}

      <Link href='#'>
        <a className='text-uppercase link-success text-decoration-none'>
          See all trends<i className='bi bi-chevron-right'></i>
        </a>
      </Link>
    </Col>
  );
}

export default Trending;
