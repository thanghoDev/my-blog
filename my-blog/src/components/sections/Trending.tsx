import React from 'react';
import Link from 'next/link';
import { Col } from 'react-bootstrap';

import Styles from '@/styles/Trending.module.css';

// types
import { Blog } from '@/types/blog';

// constants
import { CATEGORY, DETAIL } from '@/constants/pages';

type TrendingProps = {
  trending: Blog[];
};

function Trending({ trending }: TrendingProps) {
  // if (!trending) return <div>loading</div>;
  return (
    <Col md={4} className='mt-5 ps-4'>
      <h2 className='mb-5 text-decoration-underline text-capitalize fs-5'>
        trending
      </h2>

      {trending.map((item, index) => (
        //
        <div key={item.id} className='mt-3 d-flex'>
          <h3 className={Styles.number}>0{index + 1}</h3>
          <div className='ms-4'>
            <Link href={`/${DETAIL}/${item.id}`}>
              <a
                className={`text-black text-capitalize text-decoration-none ${Styles.title}`}
              >
                {item.title}
              </a>
            </Link>
            <p className='text-capitalize category mt-3'>
              Dave Rogers <code className='text-secondary'>in</code>{' '}
              <Link href={`/${CATEGORY}/${item.category}`}>
                <a className='text-black text-decoration-none'>
                  {item.category}
                </a>
              </Link>
            </p>
          </div>
        </div>
      ))}

      <Link href='#'>
        <a className='text-uppercase link-success text-decoration-none'>
          See all trends<i className='bi bi-chevron-right'></i>
        </a>
      </Link>
    </Col>
  );
}

export default Trending;
