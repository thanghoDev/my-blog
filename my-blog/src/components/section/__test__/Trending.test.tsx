import React from 'react';
import { render } from '@testing-library/react';

import Trending from '../Trending';

describe('testing Trending section', () => {
  const trending = [
    {
      id: 1,
      title: 'title',
      description: 'description',
      category: 'health',
      view: 150,
      images: 'https://i.ibb.co/10hj3h1/health.jpg',
      alt: 'health',
      content: 'content',
    },
  ];
  test('render Trending section', () => {
    const { container } = render(<Trending trending={trending} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('render fail section', () => {
    render(<Trending trending={undefined} />);
  });
});
