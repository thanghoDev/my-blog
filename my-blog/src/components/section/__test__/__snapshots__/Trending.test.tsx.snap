// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Trending section render Trending section 1`] = `
<div
  class="mt-5 ps-4 col-md-4"
>
  <h2
    class="mb-5 postTitle text-decoration-underline text-capitalize fs-5"
  >
    <a>
      trending
    </a>
  </h2>
  <div
    class="mt-3 d-flex"
  >
    <h3>
      0
      1
    </h3>
    <div
      class="ms-4"
    >
      <a
        class="text-black text-capitalize text-decoration-none undefined"
        href="/detail/1"
      >
        title
      </a>
      <p
        class="text-capitalize category mt-3"
      >
        Dave Rogers 
        <code
          class="text-secondary"
        >
          in
        </code>
         
        <a
          class="text-black text-decoration-none"
          href="/category/health"
        >
          health
        </a>
      </p>
    </div>
  </div>
  <a
    class="text-uppercase link-success text-decoration-none"
    href="/category/trending"
  >
    See all trends
    <i
      class="bi bi-chevron-right"
    />
  </a>
</div>
`;
